File: 프로그래머스/0/181868. 공백으로 구분하기 2/공백으로 구분하기 2.cpp
#include <string>
#include <vector>
#include <sstream>
#include <iostream>

using namespace std;

vector<string> solution(string my_string) {
    vector<string> answer;
    stringstream ss(my_string);
    string word;
    
    while(getline(ss, word,' ')){
        if(word != ""){
            answer.push_back(word);
            cout << word << endl;
            
        }
    }
    
    
    return answer;
}

File: 프로그래머스/0/181868. 공백으로 구분하기 2/공백으로 구분하기 2.java
import java.util.*;
class Solution {
    public String[] solution(String my_string) {

        String[] strArr = my_string.split(" ");
        
        ArrayList<String> list = new ArrayList<>();

        for(int i=0; i< strArr.length; i++){
            
            if(strArr[i].equals("")){
                continue;
            } else{
                list.add(strArr[i].trim());            
                
            }
        }

        String[] answer = new String[list.size()];
        
        for(int i=0; i< list.size(); i++){
            answer[i] = list.get(i);
        }
        
        return answer ;
    }
}